File: svws-core/src/main/java/de/svws_nrw/core/data/schoolboost/SchoolboostConnection.java
package de.svws_nrw.core.data.schoolboost;

import de.svws_nrw.transpiler.TranspilerDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Diese Klasse wird bei der Kommunikation über die Open-API-Schnittstelle verwendet.
 * Sie enthält die Informationen zu der Verbindung zu einem Schoolboost-Server.
 */
@XmlRootElement
@Schema(description = "Die Informationen zu der Verbindung zu einem Schoolboost-Server.")
@TranspilerDTO
public class SchoolboostConnection {

	/** Die ID der Verbindung. */
	@Schema(description = "die ID der Verbindung in der Datenbank. -1, wenn noch keine Verbindung eingerichtet wurde", example = "1")
	public long id = -1;

	/** Die Bezeichnung der Verbindung */
	@Schema(description = "die Bezeichnung der Verbindung", example = "Schoolboost Grundschule Musterstadt")
	public String bezeichnung = "";

	/** Die URL der Schoolboost-Anwendung. */
	@Schema(description = "Die URL der Schoolboost-Anwendung.", example = "https://app.schoolboost.de")
	public @NotNull String appUrl = "";

	/** Die URL der Schoolboost-API. */
	@Schema(description = "Die URL der Schoolboost-API.", example = "https://api.schoolboost.de/api")
	public String apiUrl = null;

	/** Die ID der Schule bei Schoolboost. */
	@Schema(description = "Die ID der Schule bei Schoolboost.", example = "b7c9d1e3-1234-5678-9abc-def012345678")
	public String schoolId = null;

	/** Gibt an, ob ein API-Key für die Verbindung hinterlegt ist. */
	@Schema(description = "gibt an, ob ein API-Key für die Verbindung hinterlegt ist.", example = "true")
	public boolean istVerbunden = false;

	/** Gibt an, ob die Schülerdaten bei der Synchronisation übertragen werden sollen. */
	@Schema(description = "gibt an, ob die Schülerdaten bei der Synchronisation übertragen werden sollen.", example = "true")
	public boolean syncSchueler = true;

	/** Gibt an, ob die Klassendaten bei der Synchronisation übertragen werden sollen. */
	@Schema(description = "gibt an, ob die Klassendaten bei der Synchronisation übertragen werden sollen.", example = "true")
	public boolean syncKlassen = true;

	/** Gibt an, ob die Lehrerdaten bei der Synchronisation übertragen werden sollen. */
	@Schema(description = "gibt an, ob die Lehrerdaten bei der Synchronisation übertragen werden sollen.", example = "true")
	public boolean syncLehrer = true;


	/**
	 * Leerer Standardkonstruktor.
	 */
	public SchoolboostConnection() {
		// leer
	}

}

File: svws-webclient/core/src/core/data/schoolboost/SchoolboostConnection.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { Class } from '../../../java/lang/Class';

export class SchoolboostConnection extends JavaObject {

	/**
	 * Die ID der Verbindung.
	 */
	public id: number = -1;

	/**
	 * Die Bezeichnung der Verbindung
	 */
	public bezeichnung: string | null = "";

	/**
	 * Die URL der Schoolboost-Anwendung.
	 */
	public appUrl: string = "";

	/**
	 * Die URL der Schoolboost-API.
	 */
	public apiUrl: string | null = null;

	/**
	 * Die ID der Schule bei Schoolboost.
	 */
	public schoolId: string | null = null;

	/**
	 * Gibt an, ob ein API-Key für die Verbindung hinterlegt ist.
	 */
	public istVerbunden: boolean = false;

	/**
	 * Gibt an, ob die Schülerdaten bei der Synchronisation übertragen werden sollen.
	 */
	public syncSchueler: boolean = true;

	/**
	 * Gibt an, ob die Klassendaten bei der Synchronisation übertragen werden sollen.
	 */
	public syncKlassen: boolean = true;

	/**
	 * Gibt an, ob die Lehrerdaten bei der Synchronisation übertragen werden sollen.
	 */
	public syncLehrer: boolean = true;


	/**
	 * Leerer Standardkonstruktor.
	 */
	public constructor() {
		super();
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.data.schoolboost.SchoolboostConnection';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.core.data.schoolboost.SchoolboostConnection'].includes(name);
	}

	public static readonly class = new Class<SchoolboostConnection>('de.svws_nrw.core.data.schoolboost.SchoolboostConnection');

	public static transpilerFromJSON(json: string): SchoolboostConnection {
		const obj = JSON.parse(json) as Partial<SchoolboostConnection>;
		const result = new SchoolboostConnection();
		if (obj.id === undefined)
			throw new Error('invalid json format, missing attribute id');
		result.id = obj.id;
		result.bezeichnung = (obj.bezeichnung === undefined) ? null : obj.bezeichnung === null ? null : obj.bezeichnung;
		if (obj.appUrl === undefined)
			throw new Error('invalid json format, missing attribute appUrl');
		result.appUrl = obj.appUrl;
		result.apiUrl = (obj.apiUrl === undefined) ? null : obj.apiUrl === null ? null : obj.apiUrl;
		result.schoolId = (obj.schoolId === undefined) ? null : obj.schoolId === null ? null : obj.schoolId;
		if (obj.istVerbunden === undefined)
			throw new Error('invalid json format, missing attribute istVerbunden');
		result.istVerbunden = obj.istVerbunden;
		if (obj.syncSchueler === undefined)
			throw new Error('invalid json format, missing attribute syncSchueler');
		result.syncSchueler = obj.syncSchueler;
		if (obj.syncKlassen === undefined)
			throw new Error('invalid json format, missing attribute syncKlassen');
		result.syncKlassen = obj.syncKlassen;
		if (obj.syncLehrer === undefined)
			throw new Error('invalid json format, missing attribute syncLehrer');
		result.syncLehrer = obj.syncLehrer;
		return result;
	}

	public static transpilerToJSON(obj: SchoolboostConnection): string {
		let result = '{';
		result += '"id" : ' + obj.id.toString() + ',';
		result += '"bezeichnung" : ' + ((obj.bezeichnung === null) ? 'null' : JSON.stringify(obj.bezeichnung)) + ',';
		result += '"appUrl" : ' + JSON.stringify(obj.appUrl) + ',';
		result += '"apiUrl" : ' + ((obj.apiUrl === null) ? 'null' : JSON.stringify(obj.apiUrl)) + ',';
		result += '"schoolId" : ' + ((obj.schoolId === null) ? 'null' : JSON.stringify(obj.schoolId)) + ',';
		result += '"istVerbunden" : ' + obj.istVerbunden.toString() + ',';
		result += '"syncSchueler" : ' + obj.syncSchueler.toString() + ',';
		result += '"syncKlassen" : ' + obj.syncKlassen.toString() + ',';
		result += '"syncLehrer" : ' + obj.syncLehrer.toString() + ',';
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

	public static transpilerToJSONPatch(obj: Partial<SchoolboostConnection>): string {
		let result = '{';
		if (obj.id !== undefined) {
			result += '"id" : ' + obj.id.toString() + ',';
		}
		if (obj.bezeichnung !== undefined) {
			result += '"bezeichnung" : ' + ((obj.bezeichnung === null) ? 'null' : JSON.stringify(obj.bezeichnung)) + ',';
		}
		if (obj.appUrl !== undefined) {
			result += '"appUrl" : ' + JSON.stringify(obj.appUrl) + ',';
		}
		if (obj.apiUrl !== undefined) {
			result += '"apiUrl" : ' + ((obj.apiUrl === null) ? 'null' : JSON.stringify(obj.apiUrl)) + ',';
		}
		if (obj.schoolId !== undefined) {
			result += '"schoolId" : ' + ((obj.schoolId === null) ? 'null' : JSON.stringify(obj.schoolId)) + ',';
		}
		if (obj.istVerbunden !== undefined) {
			result += '"istVerbunden" : ' + obj.istVerbunden.toString() + ',';
		}
		if (obj.syncSchueler !== undefined) {
			result += '"syncSchueler" : ' + obj.syncSchueler.toString() + ',';
		}
		if (obj.syncKlassen !== undefined) {
			result += '"syncKlassen" : ' + obj.syncKlassen.toString() + ',';
		}
		if (obj.syncLehrer !== undefined) {
			result += '"syncLehrer" : ' + obj.syncLehrer.toString() + ',';
		}
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

}

export function cast_de_svws_nrw_core_data_schoolboost_SchoolboostConnection(obj: unknown): SchoolboostConnection {
	return obj as SchoolboostConnection;
}
